lib/new_relic/agent/custom_event_aggregator.rb

```diff
@@ -70,15 +72,20 @@ def create_custom_event_attributes(attributes)
             key = key[0, MAX_NAME_SIZE]
           end
 
-          # value is limited to 4095
+          # value is limited to 4095 expect for LLM content-related events
           if val.is_a?(String) && val.length > MAX_ATTRIBUTE_SIZE
-            val = val[0, MAX_ATTRIBUTE_SIZE]
+            val = val[0, MAX_ATTRIBUTE_SIZE] unless llm_exempt_event_attribute?(type, key)
           end
 
           new_result[key] = val
         end
       end
 
+      def llm_exempt_event_attribute?(type, key)
+        (type == NewRelic::Agent::Llm::Embedding::EVENT_NAME && key == INPUT) ||
+          (type == NewRelic::Agent::Llm::ChatCompletionMessage::EVENT_NAME && key == CONTENT)
```

Comment:
```suggestion
        type == NewRelic::Agent::Llm::Embedding::EVENT_NAME && (key == INPUT || key == CONTENT)
```
this will slightly reduce the number of checks we need to make